### PR TITLE
Add restart feature

### PR DESCRIPTION
**Pull request by fpankretic**
https://github.com/bmwant/hapless/issues/35

### CODE REVIEW

**Comment by fpankretic** [comment]
No test were added, USAGE.md wasn't updated. If needed I can update PR with those or anything else needed.

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
```
Comment (bmwant) [inline .gitignore:130]: Usually you do not add IDE-specific files to the project-level gitignore. It should contain files related to the language/stack used within the project. 
https://github.com/github/gitignore?tab=readme-ov-file#folder-structure
If you are personally using some tools you can ignore them separately on your machine (e.g. via global ignore https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer)

File: hapless/cli.py [hunk 2]
```diff
@@ -116,6 +116,7 @@ def run(cmd, name, check):
     # NOTE: click doesn't like `required` property for `cmd` argument
     # https://click.palletsprojects.com/en/latest/arguments/#variadic-arguments
     cmd_escaped = shlex_join(cmd).strip()
+    print(cmd_escaped)
```
Comment (bmwant) [inline hapless/cli.py:119]: remove this print statement

File: hapless/main.py [hunk 3]
```diff
@@ -340,3 +357,17 @@ def signal(self, hap: Hap, sig: signal.Signals):
                 f"{config.ICON_INFO} Cannot send signal to the inactive hap",
                 style=f"{config.COLOR_ERROR} bold",
             )
+
+    def restart(self, hap: Hap):
+        hid, name, cmd = hap.hid, hap.name, hap.cmd
```
Comment (bmwant) [inline hapless/main.py:362]: maybe name should contain `-restarted` suffix or something to have some hint for the user? wdyt

File: hapless/main.py [hunk 4]
```diff
@@ -340,3 +357,17 @@ def signal(self, hap: Hap, sig: signal.Signals):
...
+
+        if hap.active:
+            self.kill([hap], verbose=False)
+
+        hap_killed = self.get_hap(str(hid))
```
Comment (bmwant) [inline hapless/main.py:367]: no need to `str`, I think it's a wrong type hint for the hid return type `-> int`

File: hapless/main.py [hunk 5]
```diff
@@ -340,3 +357,17 @@ def signal(self, hap: Hap, sig: signal.Signals):
...
+        while hap_killed.active:
+            hap_killed = self.get_hap(str(hid))
```
Comment (bmwant) [inline hapless/main.py:369]: same here, `hid` is already a str, need to add mypy linting to the CI

**Review by bmwant — CHANGES_REQUESTED** [review]
Looks fine, please add tests (at least for cli invocation with a new command) and you can update readme as well.

File: hapless/main.py (hunk 3 above)
Comment (fpankretic) [reply hapless/main.py:362]: Hmm, I would maybe personally prefer it without a suffix. If needed, some flag like `--suffix` or `--no-suffix` can be added.

**Comment by bmwant** [comment]
@fpankretic thanks for your contribution!

**Pull request merged**